src/mantine-core/src/Hr/Hr.styles.ts

```diff
@@ -17,14 +20,37 @@ export const sizes = {
   xl: 5,
 };
 
+const getInsetStyles = (inset: boolean, insetType: string) => ({
+  marginLeft: inset && (insetType === 'left' || insetType === 'middle') && 70,
+  marginRight: inset && (insetType === 'right' || insetType === 'middle') && 70,
+});
+
 export default createUseStyles(
   {
-    hr: ({ theme, size, variant, color }: HrStylesProps) => ({
+    hr: ({ theme, size, variant, color, inset, insetType }: HrStylesProps) => ({
       border: 0,
       borderTopWidth: getSizeValue({ size, sizes }),
       borderTopColor: getThemeColor({ theme, color, shade: theme.colorScheme === 'dark' ? 4 : 6 }),
       borderTopStyle: variant,
       margin: 0,
+      ...getInsetStyles(inset, insetType),
+    }),
+    hrVertical: ({ theme, size, variant, color }: HrStylesProps) => ({
+      flexDirection: 'column',
+      height: '100%',
+      borderTop: 0,
+      borderLeftWidth: getSizeValue({ size, sizes }),
+      borderLeftColor: getThemeColor({
+        theme,
+        color,
+        shade: theme.colorScheme === 'dark' ? 4 : 6,
+      }),
+      borderLeftStyle: variant,
+    }),
+    subHeader: ({ theme, color, inset, insetType }: HrStylesProps) => ({
+      fontSize: '14px',
+      color: getThemeColor({ theme, color, shade: theme.colorScheme === 'dark' ? 4 : 6 }),
```

Comment:
Same with color, it will be in subHeaderProps

Reply:
Okay

Reply:
This was just a default color, do you want me to remove this? @rtivital

Reply:
I think this would be fine?
```
 color: subHeaderProps.color
        ? subHeaderProps.color
        : getThemeColor({ theme, color, shade: theme.colorScheme === 'dark' ? 4 : 6 }),
```

Reply:
Text component already has color prop, set it with Hr color:

```tsx
<Text color={color} />
```